ts/a11y/semantic-enrich.ts

```diff
@@ -244,12 +244,21 @@ export function EnrichedMathItemMixin<N, T, D, B extends Constructor<AbstractMat
      */
     public attachSpeech(document: MathDocument<N, T, D>) {
       if (this.state() >= STATE.ATTACHSPEECH) return;
+      if (this.isEscaped || !document.options.enableEnrichment) {
+        this.state(STATE.ATTACHSPEECH);
```

Comment:
Do any of the calls below cause a MathJax retry error?  (That is, do any call `mathjax.retryAfter()`?)

If not, then you can do `this.state(STATE.ATTACHSPEECH)` above line 247, and then just return here and the other places where the state is set.  That would avoid having to set the state in several places.  But it only works if the rest of the code will always complete and not restart the typesetting via `retryAfter()`.

Reply:
It should not. The only place that triggers a `retryAfter` is in line 183:
https://github.com/mathjax/MathJax-src/blob/799ad89bcfc2dd750c7e970214d87065117a4c01/ts/a11y/semantic-enrich.ts#L183
But that should have run before `attachSpeech` at all times.